src/khoj/routers/api.py

```diff
@@ -146,9 +146,12 @@ async def search(
                     # Collate results
                     results += text_search.collate_results(hits, dedupe=dedupe)
 
+                    # Rerank results if explicitly requested or if device has GPU
+                    rerank = r or state.device.type != "cpu" or is_env_var_true("KHOJ_RERANK")
+
                     # Sort results across all content types and take top results
                     results = text_search.rerank_and_sort_results(
-                        results, query=defiltered_query, rank_results=r, search_model_name=search_model.name
+                        results, query=defiltered_query, rank_results=rerank, search_model_name=search_model.name
```

Comment:
You can see in `embeddings.py` how we determine whether to make an API query for the result or to use on-device compute. We should take that into account. Perhaps always run cross encoder if the API is being used, as one would assume the service is production-ready?

Reply:
Fair point, fixed that!